src/frontend/barnetilsyn/steg/6-vedlegg/utils.ts

```diff
@@ -58,3 +67,109 @@ export const toggleHarSendtInn = (
  */
 const dokumentajonFeltEquals = (first: DokumentasjonFelt, second: DokumentasjonFelt) =>
     first.type === second.type && first.barnId === second.barnId;
+
+/**
+ * Brukes for å opprette dokumentasjonsfelter for alle dokumentasjonsbehov funnet i tidligere steg.
+ * Samler noen dokumentasjonsbehov, slik at bruker slipper å feks laste opp samme barnehagefaktura to ganger.
+ * @returns Alle dokumentasjonsfelter bruker må fylle ut på vedleggssiden
+ */
+export const opprettDokumentasjonsfelt = (
+    dokumentasjonsbehov: Dokumentasjonsbehov[],
+    eksisterendeDokumentasjon: DokumentasjonFelt[],
+    locale: Locale
+): DokumentasjonFelt[] => {
+    const [kanSamles, kanIkkeSamles] = dokumentasjonsbehov.reduce(
+        ([kanSamles, kanIkkeSamles], currentValue) => {
+            return dokumentasjonSkalHaSamletOpplasting(currentValue)
+                ? [[...kanSamles, currentValue], kanIkkeSamles]
+                : [kanSamles, [...kanIkkeSamles, currentValue]];
+        },
+        [[], []] as [Dokumentasjonsbehov[], Dokumentasjonsbehov[]]
+    );
```

Comment:
♻️ 
Følte det var litt unødvendig komples reduce 😬 
Disse 2 skulle også kunne inlines.
```suggestion
    const kanSamles = dokumentasjonsbehov.filter(skalHaSamletOpplasting);
    const kanIkkeSamles = dokumentasjonsbehov.filter(skalIkkeHaSamletOpplasting);
    ...
    const skalIkkeHaSamletOpplasting = (dokumentasjonsbehov: Dokumentasjonsbehov) =>
    !skalHaSamletOpplasting(dokumentasjonsbehov);
```

Reply:
Ja enig, starta med en reduce -> fikk det ikke til -> gjorde det sånn som dette -> fikk til reducen og så ble den beholdt for tenkte at det ikke skulle være populært å endre sånn der. Endrer tilbake! ⭐

Reply:
Endret denne, men ikke inline for syntes det ble mer tydlig utenfor. Kan kjøre inline om det er ønskelig